Show error received by server when resolving topic from sidebar.

sahil839 opened the pull request:
<!-- Describe your pull request here.-->
- First commit is to show the error.
- Second commit is to fix alignment of "x" button.

Fixes: <!-- Issue link, or clear description.--> https://chat.zulip.org/#narrow/stream/9-issues/topic/Mark.20as.20resolve.20do.20nothing

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![chrome-capture-2023-4-24 (1)](https://github.com/zulip/zulip/assets/35494118/053c7e8d-dc37-459b-a3fe-0fab28baa206)
![chrome-capture-2023-4-26 (1)](https://github.com/zulip/zulip/assets/35494118/1f8ac9a3-8866-4019-93ba-c5e81ca043b4)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
If I'm scrolled down, the banner shows up above the visible area for me. E.g., if I'm scrolled down a little bit
![Screen Shot 2023-05-26 at 8 33 47 AM](https://github.com/zulip/zulip/assets/2090066/3aabf21e-3981-4ed0-af66-8e692d46e1a2)

(but it can also be hidden altogether)

alya commented:
> If I'm scrolled down, the banner shows up above the visible area for me.

Sounds like this comment is a general issue, and does not need to be addressed in this PR.

timabbott commented on web/src/message_edit.js:
```diff
@@ -697,7 +697,7 @@ function handle_resolve_topic_failure_due_to_time_limit(topic_is_resolved) {
 }
 
-export function toggle_resolve_topic(message_id, old_topic_name) {
+export function toggle_resolve_topic(message_id, old_topic_name, show_error_in_message_view) {
```
I think `in_message_view` doesn't feel like the right way to express the situation, since the place we display these errors is at the top of the screen; maybe something like `report_errors_in_global_banner` might make more sense?

timabbott commented:
Opened https://github.com/zulip/zulip/issues/25791 to track the error banners thing.

sahil839 commented:
Updated the PR to hide banner after 1.5 sec.
![resolve-error-1 5-sec](https://github.com/zulip/zulip/assets/35494118/70b4d9ca-f0a0-4f39-b58a-7452469855ae)

Do we want "X" even when we are hiding the banner after a short delay?

Also, adding GIF for hiding banner after 2 sec -
![resolve-error-2-sec](https://github.com/zulip/zulip/assets/35494118/1e0f6085-c25a-4c7c-9c82-5e317f4ef0de)

alya commented:
1.5s feels too fast to read to me. I like something like

>                ui_report.generic_embed_error(error_msg, 3500);

alya commented:
Keeping the `x` seems fine.

alya commented:
Can we put at `.` at the end of the message?

sahil839 commented:
Updated the PR to increase the time to 3.5 seconds and also added "." at the end of error message, but did it in the API as well since we use the same message. We are not consistent witht using "." for the error messages in the API as well currently.

![chrome-capture-2023-4-29 (2)](https://github.com/zulip/zulip/assets/35494118/d9966ce4-e75a-4dbd-a315-509b6f489cb5)

timabbott commented:
Merged, thanks @sahil839!

I think there's definitely still space for improvement here; it'd be nice if the error message said something about resolving topics.

sahil839 commented:
Yes I agree. I think adding error code will definitely help for this particular case and in general having different permissions for resolving topics would make the error messages more clear.
